### ISSUE TITLE
Actions prior to setting User ID (setUserId) "forgotten" (new Visitor ID is assigned)

### ISSUE THREAD

**Issue by peachp** [matomo-org/matomo#14709]
I saw few other similar issues, but I am not sure this is the same one.

Scenario:
1. Completely new person visits our website
2. Matomo creates records in the table `log_visit` and `log_link_visit_action` with new `idvisitor` say "0x11111111111"
3. The person performs actions A, B and C
4. The person "logs in" as person123
5. Our website calls `_paq.push(['setUserId', 'person123']);`
...at this stage Matomo creates another record in `log_link_visit_action` and `log_visit`, but with new `idvisitor`, say **0x22222222** and with `user_id` **person123**
6. User **person123** / visitor **0x22222222** performs actions D and E.

Issue: the profile for the user **person123** in Matomo only shows actions D and E.
Expectation: the profile for the user **person123** should show actions A, B, C, D and E.

In other words, I don't see a reason why Matomo starts seeing the same person as new visitor once she "logged in" (= User ID is set via JS tracking API).

**Comment by tsteur** [comment]
Thanks for creating the issue 👍 We will be changing this in https://github.com/matomo-org/matomo/pull/14360

**Comment by peachp** [comment]
OK, thanks for pointing to it
